weave-js/track-v1.js

```diff
@@ -0,0 +1,185 @@
+(async function () {
+  // OneTrust Cookies Consent Notice
+  // OneTrust requires JQuery, and by default loads an outdated version along with its own script.
+  // We're turning off that default behavior and loading our own updated version of JQuery instead.
+  // We must ensure that JQuery is fully loaded before inserting the OneTrust script
+  // to avoid potential timing issues.
+  insertScript('/jquery-3.6.0.min.js').then(() =>
+    insertScript(
+      'https://cdn.cookielaw.org/consent/da0f7ab4-835a-42da-beeb-767dcff7d1cc.js'
+    )
+  );
+
+  function insertCookieControlledScript(
+    url,
+    cookieControlGroupID,
+    callback,
+    async
+  ) {
+    // Use Optanon to conditionally insert tracking scripts. This is the "Onetrust"
+    // cookie consent platform (https://app.onetrust.com) that we use to get
+    // cookie consent from European users. Cookies can be enabled/disabled by
+    // category. Categories are configured in the OneTrust UI. cookieControlGroupID
+    // specifies the category.
+
+    window.Optanon.InsertScript(
+      url,
+      'body',
+      callback,
+      null,
+      cookieControlGroupID,
+      async || false
+    );
+  }
+
+  function addSegmentAnalyticsMethods() {
+    var analytics = (window.analytics = window.analytics || []);
+    // Segment boilerplate. Add stubs for segment analytics methods so
+    // they can be called even if segment is disabled.
+    if (analytics.invoked) {
+      console.error('Segment snippet included twice.');
+    } else {
+      analytics.invoked = !0;
+      analytics.methods = [
```

Comment:
do we need all these? What does this list do?

Reply:
See comment above:

```
// Segment boilerplate. Add stubs for segment analytics methods so
// they can be called even if segment is disabled.
```